### PR TITLE
Implement geopos reply helper function

### LINKED ISSUE #230: Add reply helper functions for GEO commands
Add [reply helper functions](https://godoc.org/github.com/garyburd/redigo/redis#hdr-Reply_Helpers) for the GEO commands. 

Note: Redigo supports all GEO commands. This issue is about adding convenience functions for working with GEO command results.

### PR DESCRIPTION
**Pull request by crackerplace**
Implemented a helper function for geopos reply.

@garyburd 

Fixes #230

### CODE REVIEW

**Comment by garyburd** [comment]
The function should handle null values without returning an error.  One way is to return a slice of pointers to lat/long pairs:

     func Positions(result interface{}, err error) ([]*[2]float64, error)

**Comment by crackerplace** [comment]
@garyburd done.

File: redis/reply.go [hunk 1]
```diff
@@ -394,13 +394,16 @@ func Int64Map(result interface{}, err error) (map[string]int64, error) {
...
 		return nil, err
 	}
-	positions := make([][2]float64, len(values))
+	positions := make([]*[2]float64, len(values))
 	for i := range values {
+		if values[i] == nil {
+			continue
+		}
 		p, ok := values[i].([]interface{})
 		if !ok {
```
Comment (garyburd) [inline redis/reply.go]: Also check len(p) >= 2

Comment (crackerplace) [inline redis/reply.go]: If the position for a given input member is returned in the response,do we have to even check if the position contains both latitude and longitude ? Does geoadd command allow not specifying either latitude or longitude for a given member.

Comment (garyburd) [inline redis/reply.go]: Yes. The function should return an error in the case of an unexpected response. The function should not panic.

Comment (crackerplace) [inline redis/reply.go]: ok.In that case,I will return error if if len(p) != 2.

**Comment by crackerplace** [comment]
@garyburd Handled the length of the position array.

File: redis/reply.go [hunk 2]
```diff
@@ -391,3 +391,35 @@ func Int64Map(result interface{}, err error) (map[string]int64, error) {
...
+	positions := make([]*[2]float64, len(values))
+	for i := range values {
+		if values[i] == nil {
+			continue
+		}
+		p, ok := values[i].([]interface{})
+		if len(p) != 2 {
+			return nil, fmt.Errorf("redigo: unexpected number of values for a member position,got %d", len(p))
+		}
+		if !ok {
```
Comment (garyburd) [inline redis/reply.go]: The true branch of this if statement will never be executed because len(p) != 2 if ok == false.

Comment (crackerplace) [inline redis/reply.go]: @garyburd yes,I moved it to the right position.thanks.

**Comment by crackerplace** [comment]
@garyburd squashed the commits.thanks.

**Pull request merged**